unity-renderer/Assets/UIComponents/Tests/ProfileCardComponentViewTests.cs

```diff
@@ -0,0 +1,159 @@
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ProfileCardComponentViewTests
+{
+    private ProfileCardComponentView profileCardComponent;
+
+    [SetUp]
+    public void SetUp() { profileCardComponent = BaseComponentView.Create<ProfileCardComponentView>("ProfileCard"); }
+
+    [TearDown]
+    public void TearDown()
+    {
+        profileCardComponent.Dispose();
+        GameObject.Destroy(profileCardComponent.gameObject);
+    }
+
+    [Test]
+    public void SetOnClickCorrectly()
+    {
+        // Arrange
+        bool isClicked = false;
+        profileCardComponent.onClick.AddListener(() => isClicked = true);
+
+        // Act
+        profileCardComponent.button.onClick.Invoke();
+
+        // Assert
+        Assert.IsTrue(isClicked, "The button has not responded to the onClick action.");
+    }
+
+    [Test]
+    public void ConfigureButtonCorrectly()
+    {
+        // Arrange
+        ProfileCardComponentModel testModel = new ProfileCardComponentModel
+        {
+            profilePicture = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero),
+            profileName = "Test name",
+            profileAddress = "Test address",
+            onClickEvent = new Button.ButtonClickedEvent()
+        };
+
+        // Act
+        profileCardComponent.Configure(testModel);
+
+        // Assert
+        Assert.AreEqual(testModel, profileCardComponent.model, "The model does not match after configuring the button.");
+    }
+
+    [Test]
+    public void RefreshButtonCorrectly()
+    {
+        // Arrange
+        Sprite testPicture = Sprite.Create(new Texture2D(10, 10), new Rect(), Vector2.zero);
```

Comment:
The `Texture2D` objects have to be destroyed manually to avoid leaks.

Reply:
Good catch! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/531831692da512ed2d0095f4e2c0d08be70ee7be